src/dmd/opover.d

```diff
@@ -770,16 +781,20 @@ extern (C++) Expression op_overload(Expression e, Scope* sc)
                     return;
                 }
                 // Didn't find it. Forward to aliasthis
-                if (ad.aliasthis)
+                if (!e.aliasthislock)
                 {
                     /* Rewrite op(e1) as:
-                     *      op(e1.aliasthis)
+                     *  op(e1.aliasthis)
                      */
-                    Expression e1 = new DotIdExp(e.loc, e.e1, ad.aliasthis.ident);
-                    result = e.copy();
-                    (cast(UnaExp)result).e1 = e1;
-                    result = result.trySemantic(sc);
-                    return;
+
+                    Expression[] results;
+                    iterateAliasThis(sc, e.e1, &UnaAliasThisCtx(e).atUna, results);
+                    Expression ret = enforceOneResult(results, e.loc, "unable to unambiguously resolve %s; candidates:", e.toChars());
```

Comment:
I may create a function with ton of arguments which combine `iterateAliasThis` and `enforceOneResult`, but I don't want to do that, because `iterateAliasThis` has a default values for the last arguments and `enforceOneResult` has a variadic arguments.
And they do independent things:
`iterateAliasThis` collects candidates to resolve,
`enforceOneResult` checks that there is only one candidate and raises an error otherwise.
And `enforceOneResult` and take a different error messages. They similar but not the same.
What do you think?

Reply:
If you can't find a way to consolidate the context so you don't need to create a function with too many arguments, I'd say it's not worth it, but I hope there's a way.